### PR TITLE
feat: Introduce gateway template

### PR DESCRIPTION
**Pull request by akim-bow**

### CODE REVIEW

File: src/lib/init.ts [hunk 1]
```diff
@@ -523,3 +679,66 @@ function getPackageJSON(isJS: boolean) {
...
+    keywords: [],
+    scripts: {
+      ...(isJS
+        ? {}
+        : {
+            build: "rm -rf ./dist && tsc -p tsconfig.json",
+          }),
+      start: isJS
+        ? "node src/server.js"
+        : "node --loader ts-node/esm src/server.ts",
```
Comment (shamsartem) [inline src/lib/init.ts:699]:
```suggestion
        ? "node --no-warnings src/server.js"
        : "node --no-warnings --loader ts-node/esm src/server.ts",
```

File: src/lib/init.ts [hunk 2]
```diff
@@ -523,3 +679,66 @@ function getPackageJSON(isJS: boolean) {
...
+      "@sinclair/typebox": "0.32.11",
+    },
+    devDependencies: isJS
+      ? {}
+      : {
+          "@types/node": "20.11.5",
+          "@fastify/type-provider-typebox": "4.0.0",
+          typescript: "5.3.3",
+          "ts-node": "10.9.2",
+        },
```
Comment (shamsartem) [inline src/lib/init.ts:715]: also typescript can be updated in the frontend template
```suggestion
    ...(isJS
      ? {}
      : {
          devDependencies: {
            "@types/node": "20.11.5",
            "@fastify/type-provider-typebox": "4.0.0",
            typescript: "5.3.3",
            "ts-node": "10.9.2",
          },
        }),
```

Comment (akim-bow) [reply src/lib/init.ts:715]: Didn't understand your code suggestion. There is no need to spread anything i believe. Or what's the reasoning behind this?

File: src/lib/multiaddres.ts [hunk 3]
```diff
@@ -398,30 +398,35 @@ export async function updateRelaysJSON({
...
+      commandObj.error(
+        `relaysPath must be relative to the root project directory. Found: ${relayPath} at ${fluenceConfig.$getPath()}}`,
+      );
+    }
+
+    await writeFile(
+      join(resolve(projectRootDir, relayPath), "relays.json"),
+      jsonStringify(relays),
+    );
+  }
```
Comment (shamsartem) [inline src/lib/multiaddres.ts:429]: A small improvement can be to use `splitErrorsAndResults` to find ALL absolute paths at once so you can show all of them at the same time to the user like this
`relayPath must contain only paths relative to the root project directory. Found: ${absolutePaths.join('\n')}`
and then do `await Promise.all(relativePaths.map((relativePath) => writeFile(...)))`

File: src/lib/init.ts (hunk 2 above)
Comment (shamsartem) [reply src/lib/init.ts]: It's just that if you don't have any `devDependencies` in js template there is no need to have an empty `devDependencies: {}` in package.json there. That's it

File: src/lib/multiaddres.ts (hunk 3 above)
Comment (akim-bow) [reply src/lib/multiaddres.ts]: Made the suggested improvement. Please check

File: src/lib/init.ts (hunk 2 above)
Comment (akim-bow) [reply src/lib/init.ts]: Oh, so it's about empty `devDependencies`... Yeap, nice. Seems like i misread it the first time.

File: src/lib/init.ts [hunk 4]
```diff
@@ -341,6 +332,58 @@ async function initTSorJSProject({
...
+        compileArgs: {
+          filePath: await ensureAquaMainPath(),
+          imports: await getAquaImports({
+            maybeFluenceConfig: fluenceConfig,
+          }),
+        },
+        targetType: isJS ? "js" : "ts",
+        outputPath: gatewayCompiledAquaPath,
+      });
+    })(),
```
Comment (shamsartem) [inline src/lib/init.ts:383]: can it be just?
```suggestion
    compileToFiles({
        compileArgs: {
          filePath: await ensureAquaMainPath(),
          imports: await getAquaImports({
            maybeFluenceConfig: fluenceConfig,
          }),
        },
        targetType: isJS ? "js" : "ts",
        outputPath: gatewayCompiledAquaPath,
      }),
```

Comment (shamsartem) [reply src/lib/init.ts:383]: and above as well. Don't know why it was like that

Comment (shamsartem) [reply src/lib/init.ts:383]: it's fine as it is I guess

**Review by shamsartem — APPROVED** [review]

File: src/lib/init.ts [hunk 5]
```diff
@@ -519,7 +720,78 @@ function getPackageJSON(isJS: boolean) {
...
+
+Here you can find an example of simple Fluence HTTP gateway.
+The gateway allows you to map Aqua functions to http requests i.e., you can create HTTP route and handle the incoming request by executing some Aqua function.
+You can check it out and test this repo.
+
+### Start gateway
+
+- \`npm install\`
+- \`npm run start\`
+- \`curl -X POST http://localhost:8080/my/callback/hello -H "ACCESS_TOKEN: abcdefhi" -H 'Content-Type: application/json' -d '{"name": "Fluence" }'\`
```
Comment (folex) [inline src/lib/init.ts:782]: Let's also explain that there are `runDeployedServices` and `showSubnet` that correspond to `fluence run -f runDeployedServices` and `fluence run -f showSubnet`, and they work after `deal deploy`?

**Review by folex — APPROVED** [review]

**Pull request merged**